apps/backend/src/datasources/postgres/InstrumentDataSource.ts

```diff
@@ -667,6 +667,48 @@ export default class PostgresInstrumentDataSource
     );
   }
 
+  async unsubmitInstrument(
+    proposalPks: number[],
+    instrumentId: number
+  ): Promise<InstrumentsHasProposals> {
+    const records: FapProposalRecord[] = await database('fap_proposals')
+      .update(
+        {
+          fap_meeting_instrument_submitted: false,
+        },
+        ['*']
+      )
+      .whereIn('proposal_pk', proposalPks)
+      .andWhere('instrument_id', instrumentId);
+
+    if (!records?.length) {
+      throw new GraphQLError(
+        `Some record from fap_proposals not found with proposalPks: ${proposalPks} and instrumentId: ${instrumentId}`
+      );
+    }
+
+    const instrumentHasProposls = await this.getInstrumentHasProposals(
+      instrumentId,
+      proposalPks
+    );
+
+    if (
+      instrumentHasProposls.instrumentHasProposalIds.length !==
+      proposalPks.length
+    ) {
+      throw new GraphQLError(
```

Comment:
Maybe this is a little unexpected that this throws an error, because it could be that some rows actually got updated.
We could either put this in the transaction and rollback or just add the log and return succesfully updated records.

Reply:
Nice point. I will take a look at this and see what can we do.